Common/cpp/Tools/RuntimeDecorator.cpp

```diff
@@ -0,0 +1,77 @@
+//
+//  RuntimeDecorator.cpp
+//  DoubleConversion
+//
+//  Created by Szymon Kapala on 31/03/2020.
+//
+
+#include "RuntimeDecorator.h"
+#include <unordered_map>
+#include <memory>
+#include "Logger.h"
+
+void RuntimeDecorator::addGlobalMethods(jsi::Runtime &rt) {
+  std::unordered_map<std::string, jsi::Value> properties;
+  
+  // add assign method
+  std::string assignCode = R"((function assign(left, right) {
+  if ((typeof right === 'object') && (!right.value)) {
+    for (let key of Object.keys(right)) {
+      if (left[key]) {
+        assign(left[key], right[key]);
+      }
+    }
+
+  } else if (Array.isArray(right)) {
+    for (let i; i < right.length; i++) {
+      assign(left[i], right[i]);
+    }
+
+  } else {
+    if (left.set) {
+      if (right.value) {
+        left.set(right.value);
+      } else {
+        left.set(right);
+      }
+    }
+  }}))";
+  properties["assign"] = rt.evaluateJavaScript(std::make_shared<jsi::StringBuffer>(assignCode), "");
+  
+  // event worklet constants
+  properties["START"] = jsi::Value(2);
+  properties["END"] = jsi::Value(5);
+  
+  class Animated : public jsi::HostObject {
+    std::unordered_map<std::string, jsi::Value> props;
+  public:
+    Animated(std::unordered_map<std::string, jsi::Value> && props) {
+      this->props = std::move(props);
+    }
+
+    jsi::Value get(jsi::Runtime &rt, const jsi::PropNameID &name) {
+      auto propName = name.utf8(rt);
+      
+      if (props.count(propName) > 0) {
+        return jsi::Value(rt,props[propName]);
```

Comment:
You search the collection twice for the same key. I'd go with something like: `auto it = props.find(propName); if(it != props.end()) return jsi::Value(rt, *it);` (treat it as a pseudocode pls)

Reply:
Done